src/panels/config/devices/device-detail/ha-device-entities-card.ts

```diff
@@ -92,36 +93,43 @@ export class HaDeviceEntitiesCard extends LitElement {
 
     return html`
       <ha-card outlined .header=${this.header}>
-        <div id="entities">
-          <mwc-list>
-            ${shownEntities.map((entry) =>
-              this.hass.states[entry.entity_id]
-                ? this._renderEntity(entry)
-                : this._renderEntry(entry)
-            )}
-          </mwc-list>
-        </div>
-        ${hiddenEntities.length
-          ? !this.showHidden
-            ? html`
-                <button class="show-more" @click=${this._toggleShowHidden}>
-                  ${this.hass.localize(
-                    "ui.panel.config.devices.entities.hidden_entities",
-                    { count: hiddenEntities.length }
-                  )}
-                </button>
-              `
-            : html`
+        ${shownEntities.length
+          ? html`
+              <div id="entities" class="move-up">
                 <mwc-list>
-                  ${hiddenEntities.map((entry) => this._renderEntry(entry))}
-                </mwc-list>
-                <button class="show-more" @click=${this._toggleShowHidden}>
-                  ${this.hass.localize(
-                    "ui.panel.config.devices.entities.show_less"
+                  ${shownEntities.map((entry) =>
+                    this.hass.states[entry.entity_id]
+                      ? this._renderEntity(entry)
+                      : this._renderEntry(entry)
                   )}
-                </button>
-              `
-          : ""}
+                </mwc-list>
+              </div>
+            `
+          : nothing}
+        ${hiddenEntities.length
+            ? html`<div class=${classMap({ "move-up": !shownEntities.length })}>
```

Comment:
You need to import `classMap`!

`import { classMap } from "lit/directives/class-map";`

Reply:
Thanks, this last addition was not tested locally(